plaid/assets.go

```diff
@@ -3,25 +3,90 @@ package plaid
 import (
 	"encoding/json"
 	"errors"
+	"time"
 )
 
 type AssetReport struct {
 	AssetReportID  string            `json:"asset_report_id"`
 	ClientReportID string            `json:"client_report_id"`
-	DateGenerated  string            `json:"date_generated"`
+	DateGenerated  time.Time         `json:"date_generated"`
```

Comment:
DateGenerated and DateLastUpdatedAt are datetimes with the format "2018-04-12T03:32:11Z". `time. RFC3339` 

Instead of overriding the marshaller for each type that contains a date/datetime if it would be better to create a type alias for date and datetime respectfully and implement the marshaller there.

Reply:
@cbelsole something like this?

``` go
type Time struct {
    time.Time
}

func (t Time) MarshalJSON() ([]byte, error) { /* ... */ }
```

Reply:
```golang
type (
	Date time.Time
 	DateTime time.Time
)

func (t Date) MarshalJSON() ([]byte, error) { /* ... */ }
func (t DateTime) MarshalJSON() ([]byte, error) { /* ... */ }

type AssetReport struct {
	DateGenerated DateTime `json:"date_generated"`
}
```

Reply:
If we do that, clients will need to cast to/from `time.Time` everywhere.

``` go
report := plaid.AssetReport{
    DateGenerated: plaid.Date(time.Now())
}

time.Time(report.DateGenerated).Unix()
```

With the embedded type, you can still access all the methods.

Reply:
agreed

Reply:
@cbelsole so which approach do you want to take?

Reply:
defining a `plaid.Date` and `plaid.DateTime` seems reasonable to me. 

we actually have these defined internally. let me open a PR to add them, and then you can use them here after rebasing @cbelsole @icholy